libs/vkd3d/command.c

```diff
@@ -2671,96 +2645,76 @@ static void d3d12_command_list_clear_attachment_pass(struct d3d12_command_list *
-    extent.height = d3d12_resource_desc_get_height(&resource->desc, view->info.texture.miplevel_idx);
-    extent.depth = view->info.texture.layer_count;
+    if (separate_ds_layouts)
+    {
+        image_barriers[0].subresourceRange.aspectMask = VK_IMAGE_ASPECT_DEPTH_BIT;
+
+        image_barriers[1] = image_barriers[0];
+        image_barriers[1].newLayout = stencil_attachment_info.imageLayout;
+        image_barriers[1].subresourceRange.aspectMask = VK_IMAGE_ASPECT_STENCIL_BIT;
+        image_barrier_count = 2;
+    }
+    else
+        image_barrier_count = 1;
 
-    if (!d3d12_command_list_create_framebuffer(list, vk_render_pass, 1, &view->vk_image_view, extent, &vk_framebuffer))
+    for (i = 0; i < image_barrier_count; i++)
     {
-        ERR("Failed to create framebuffer.\n");
-        return;
+        image_barriers[i].srcAccessMask = clear_op ? access : 0;
+        image_barriers[i].dstAccessMask = access;
+        image_barriers[i].oldLayout = initial_layouts[i];
     }
 
-    subpass_begin_info.sType = VK_STRUCTURE_TYPE_SUBPASS_BEGIN_INFO_KHR;
-    subpass_begin_info.pNext = NULL;
-    subpass_begin_info.contents = VK_SUBPASS_CONTENTS_INLINE;
+    VK_CALL(vkCmdPipelineBarrier(list->vk_command_buffer,
+        stages, stages, 0, 0, NULL, 0, NULL,
+        image_barrier_count, image_barriers));
 
-    begin_info.sType = VK_STRUCTURE_TYPE_RENDER_PASS_BEGIN_INFO;
-    begin_info.pNext = NULL;
-    begin_info.renderPass = vk_render_pass;
-    begin_info.framebuffer = vk_framebuffer;
-    begin_info.renderArea.offset.x = 0;
-    begin_info.renderArea.offset.y = 0;
-    begin_info.renderArea.extent.width = extent.width;
-    begin_info.renderArea.extent.height = extent.height;
-    begin_info.clearValueCount = clear_op ? 1 : 0;
-    begin_info.pClearValues = clear_op ? clear_value : NULL;
-
-    VK_CALL(vkCmdBeginRenderPass2KHR(list->vk_command_buffer,
-            &begin_info, &subpass_begin_info));
+    rendering_info.renderArea.offset.x = 0;
+    rendering_info.renderArea.offset.y = 0;
+    rendering_info.renderArea.extent.width = d3d12_resource_desc_get_width(&resource->desc, view->info.texture.miplevel_idx);
+    rendering_info.renderArea.extent.height = d3d12_resource_desc_get_height(&resource->desc, view->info.texture.miplevel_idx);
+    rendering_info.layerCount = view->info.texture.layer_count;
+
+    VK_CALL(vkCmdBeginRenderingKHR(list->vk_command_buffer, &rendering_info));
 
     if (!clear_op)
     {
         d3d12_command_list_clear_attachment_inline(list, resource, view, 0,
                 clear_aspects, clear_value, rect_count, rects);
     }
 
-    subpass_end_info.sType = VK_STRUCTURE_TYPE_SUBPASS_END_INFO_KHR;
-    subpass_end_info.pNext = NULL;
+    VK_CALL(vkCmdEndRenderingKHR(list->vk_command_buffer));
 
-    VK_CALL(vkCmdEndRenderPass2KHR(list->vk_command_buffer, &subpass_end_info));
+    for (i = 0; i < image_barrier_count; i++)
```

Comment:
final_layouts should always be equal to attachment layout I think, so there should be no need to do barriers here.

Technically, I suppose we would need access -> access flush for COLOR_WRITE | DEPTH_WRITE, etc, but we don't do that for normal render passes due to the massive slowdown.

Reply:
Fixed.